src/index/CompressedRelation.cpp

```diff
@@ -1179,14 +1179,34 @@ auto CompressedRelationReader::getFirstAndLastTriple(
     return {row[0], row[1], row[2], row[ADDITIONAL_COLUMN_GRAPH_ID]};
   };
 
-  auto firstBlock = scanBlock(blocks.front());
```

Comment:
I now get why the code is written the way it is, but it wasn't immediately obvious to me. The code below is guaranteed to never read any block more than once, and assumes very few blocks will actually be empty, so it is continuing to read from the end of the list once a first non-empty block is found. Please comment that for future readers.